src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala

```diff
@@ -995,10 +998,14 @@ abstract class GenASM extends SubComponent with BytecodeWriters { self =>
      * Invoked from genMethod() and addForwarder().
      */
     def addRemoteExceptionAnnot(isRemoteClass: Boolean, isJMethodPublic: Boolean, meth: Symbol) {
+      def hasThrowsRemoteException = meth.annotations.exists {
+        case ThrownException(exc) => exc.typeSymbol == definitions.RemoteExceptionClass
+        case _ => false
+      }
       val needsAnnotation = (
         (  isRemoteClass ||
            isRemote(meth) && isJMethodPublic
-        ) && !(meth.throwsAnnotations contains RemoteExceptionClass)
+        ) && !hasThrowsRemoteException
```

Comment:
@retronym A test in the GenASM-removal change depends on this fix, so without it, my change can't be merged.
@lrytz You could also just ignore the change in GenASM, considering that it's going away.